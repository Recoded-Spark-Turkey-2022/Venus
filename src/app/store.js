// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
