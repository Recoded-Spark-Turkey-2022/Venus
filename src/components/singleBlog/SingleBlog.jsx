import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import {
  AiOutlineShareAlt,
  AiFillFacebook,
  AiOutlineTwitter,
  AiFillHeart,
} from 'react-icons/ai';
import { TfiCommentsSmiley } from 'react-icons/tfi';
import { FaCalendarAlt } from 'react-icons/fa';
import { Flip, toast, ToastContainer } from 'react-toastify';
import {
  collection,
  doc,
  getDocs,
  query,
  updateDoc,
  where,
} from 'firebase/firestore';
import { useParams } from 'react-router-dom';

import Container from '../UI/Container';
import { getAllListings } from '../../features/listings/listingSlice';
import BlogItem from '../blogs/BlogItem';
import './singleBlog.css';

import { db } from '../../firebase/firebase.config';

const SingleBlog = ({ data }) => {
  const {
    avatars,
    title,
    userName,
    text,
    content,
    ImageUrl,
    timeStamp,
    upVote: vote,
  } = data[0];
  const params = useParams();
  const dataStore = useSelector(getAllListings);
  const [tooltip, setTooltip] = useState(false);
  const [docId, setDocId] = useState('');
  const [upVote, setVote] = useState(vote);
  const [date, setDate] = useState('');
  const randomNumber = Math.floor(Math.random() * dataStore.length) - 1;
  const randomNum1 = randomNumber > 0 ? randomNumber : 0;
  const randomNum2 = randomNumber > 0 ? randomNumber + 1 : 5;
  const randomArr = [];
  randomArr.push(dataStore[randomNum1]);
  randomArr.push(dataStore[randomNum2]);

  const { seconds } = timeStamp;

  useEffect(() => {
    const output = new Date(seconds * 1000).toLocaleDateString();
    setDate(output);
    setVote(vote);
    setTooltip(false);
  }, [seconds]);
  useEffect(() => {
    const fetchListingData = async () => {
      try {
        const listingRef = collection(db, 'listings');

        const queryOflist = query(
          listingRef,
          where('userRef', '==', params.blogId)
        );

        const querySnap = await getDocs(queryOflist);

        querySnap.forEach((document) => {
          setDocId(document.id);
        });

        if (!querySnap) {
          console.log('Could not fetch the data');
        }
      } catch (error) {
        console.log(error.message);
      }
    };

    fetchListingData();
  }, [params.blogId]);

  const currentUrl = window.location.href;
  const handleCopy = () => {
    navigator.clipboard.writeText(window.location.href);

    toast.success('Link Coppied!', {
      position: 'top-left',
      autoClose: 1200,
      hideProgressBar: false,
      closeOnClick: true,
      pauseOnHover: true,
      draggable: true,
      progress: undefined,
      theme: 'light',
    });
  };
  const shareOnFacebook = () => {
    const facebookShare = 'https://www.facebook.com/sharer/sharer.php?u=';
    const navUrl = facebookShare + currentUrl;

    window.open(navUrl, '_blank');
  };
  const shareOnTwitter = () => {
    const twitterShare = 'https://twitter.com/intent/tweet?text=';
    const navUrl = twitterShare + currentUrl;

    window.open(navUrl, '_blank');
  };
  const isAuthenticated = true;

  const handleLike = () => {
    if (isAuthenticated) {
      toast('Thank You!', {
        position: 'top-left',
        icon: <TfiCommentsSmiley className=" text-rose text-3xl " />,
        autoClose: 1200,

        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        hideProgressBar: true,
        theme: 'light',
        className: 'thanks-tooltip',
      });
    }

    setVote(upVote + 1);

    setTooltip(true);
  };
  useEffect(() => {
    const updateLike = async () => {
      try {
        if (docId) {
          const docRef = doc(db, 'listings', docId);

          await updateDoc(docRef, { upVote });
        }
      } catch (error) {
        console.log(error.message);
      }
    };
    updateLike();
  }, [upVote]);

  return (
    <Container>
      <div className="flex mt-[60px] flex-col gap-10 pb-0  md:pb-24 md:flex-row main-container">
        <article className="flex flex-col w-full md:w-[60%] relative">
          <h1 className="text-3xl uppercase font-medium text-center mb-4">
            {title}{' '}
          </h1>
          <div>
            <img
              className="w-[100%] object-cover rounded-xl"
              src={ImageUrl}
              alt={title}
            />
          </div>
          <div className="flex items-center justify-around my-2">
            <p className="flex  align-middle self-center justify-center items-end ">
              <span>by</span>{' '}
              <img
                className="w-[35px] h-[35px] rounded-full"
                src={avatars[0]}
                alt="avatar"
              />{' '}
              <span> {userName} </span>
            </p>
            <div className=" flex  gap-5 ">
              <AiOutlineShareAlt
                onClick={handleCopy}
                className="text-darkBlue textShadow  text-3xl font-bold duration-300 brightness-110 hover:brightness-50 cursor-pointer"
              />
              <AiFillFacebook
                onClick={shareOnFacebook}
                className="text-darkBlue text-3xl font-bold duration-300 brightness-110 hover:brightness-50 cursor-pointer"
              />
              <AiOutlineTwitter
                onClick={shareOnTwitter}
                className="text-darkBlue text-3xl font-bold duration-300 brightness-110 hover:brightness-50 cursor-pointer"
              />
            </div>
            <div className="date flex gap-3 items-center">
              <FaCalendarAlt className="text-darkBlue textShadow  self-baseline text-3xl font-bold duration-300 brightness-110 hover:brightness-50" />
              <span className="self-baseline">{date} </span>
            </div>
          </div>
          <div>
            <h2 className="text-xl font-medium my-3">{text} </h2>
            <p>{content} </p>
          </div>
          <button
            type="button"
            disabled={tooltip}
            onClick={handleLike}
            className="heart-container shadow-md flex items-center justify-center absolute bottom-[-60px] right-5"
          >
            <AiFillHeart className="text-rose text-2xl mr-1 duration-500 brightness-125 hover:brightness-75" />
            <span className="text-lg font-medium">{upVote}</span>
          </button>
        </article>
        <aside className="w-full md:w-[40%] h-fit mt-20 md:mt-0 ">
          <h2 className="text-2xl font-medium text-center mb-4">Read Also</h2>
          <div className="flex blog-aside flex-wrap justify-center gap-10 w-full blog-wrapper">
            {randomArr?.map((singlePost) => (
              <BlogItem
                key={singlePost.userRef}
                {...singlePost}
                // eslint-disable-next-line react/jsx-boolean-value
                fullWidth={true}
              />
            ))}
          </div>
        </aside>
      </div>
      <ToastContainer transition={Flip} limit={3} />
    </Container>
  );
};

export default SingleBlog;
