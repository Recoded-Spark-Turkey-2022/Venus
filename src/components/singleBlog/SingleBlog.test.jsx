import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { store } from '../../app/store';
import SingleBlog from './SingleBlog';


const props = {
    avatars: 'test',
    title: 'test',
    userName: 'test',
    text: 'test',
    content: 'test',
    ImageUrl: 'test',
    timeStamp: 'test',
    userRef:"test",
    upVote: 'test'
  };

describe('SingleBlog', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
          <Router>
            <Provider store={store}>
              <SingleBlog data={props} />
            </Provider>
          </Router>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });