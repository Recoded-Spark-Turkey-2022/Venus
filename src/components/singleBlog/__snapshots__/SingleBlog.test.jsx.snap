// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleBlog renders correctly 1`] = `
<div
  className="container mx-auto px-10"
  data-testid="container"
>
  <div
    className="flex mt-[100px] flex-col gap-10 pb-0  md:pb-24 md:flex-row main-container"
  >
    <article
      className="flex flex-col w-full md:w-[60%] relative"
    >
      <h1
        className="text-3xl uppercase font-medium text-center mb-4"
      >
         
      </h1>
      <div>
        <img
          className="w-[100%] object-cover rounded-xl"
        />
      </div>
      <div
        className="flex flex-col sm:flex-row  gap-5 sm:gap-1 items-center justify-around my-2"
      >
        <p
          className="flex  align-middle self-center justify-center items-end "
        >
          <span>
            by
          </span>
           
          <img
            alt="avatar"
            className="w-[35px] h-[35px] rounded-full"
          />
           
          <span>
             
             
          </span>
        </p>
        <div
          className="flex gap-3 sm:gap-1 items-end"
        >
          <div
            className=" flex  gap-5  mr-2"
          >
            <svg
              className="text-darkBlue textShadow  text-3xl font-bold duration-300 brightness-110 hover:brightness-50 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 0 0 0-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
              />
            </svg>
            <svg
              className="text-darkBlue text-3xl font-bold duration-300 brightness-110 hover:brightness-50 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
              />
            </svg>
            <svg
              className="text-darkBlue text-3xl font-bold duration-300 brightness-110 hover:brightness-50 cursor-pointer"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0 0 75-94 336.64 336.64 0 0 1-108.2 41.2A170.1 170.1 0 0 0 672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 0 0-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 0 1-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 0 1-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
              />
            </svg>
          </div>
          <div
            className="date flex gap-3 items-end"
          >
            <svg
              className="text-darkBlue textShadow  self-end text-3xl font-bold duration-300 brightness-110 hover:brightness-50"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
              />
            </svg>
            <span
              className="self-end"
            >
               
            </span>
          </div>
        </div>
      </div>
      <div>
        <h2
          className="text-xl font-medium my-3"
        >
           
        </h2>
        <p>
           
        </p>
      </div>
      <button
        className="heart-container shadow-md flex items-center h-[43px] justify-center absolute bottom-[-60px] right-5"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          className="text-rose text-2xl mr-1 duration-500 brightness-125 hover:brightness-75"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
          />
        </svg>
        <span
          className="text-lg font-medium"
        />
      </button>
      <button
        className="heart-container shadow-md flex items-center  h-[43px] justify-center absolute bottom-[-60px] right-[6rem]"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="text-[#ff3000] text-2xl mr-1 duration-500 brightness-125 hover:brightness-75"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M292.7 840h438.6l24.2-512h-487z"
          />
          <path
            d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-504-72h304v72H360v-72zm371.3 656H292.7l-24.2-512h487l-24.2 512z"
          />
        </svg>
      </button>
      <button
        className="heart-container shadow-md flex items-center   h-[43px] justify-center absolute bottom-[-60px] right-[10.3rem]"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="text-[#ff8239] font-bolder text-2xl mr-1 duration-500 "
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="Edit"
          >
            <g>
              <path
                d="M3.548,20.938h16.9a.5.5,0,0,0,0-1H3.548a.5.5,0,0,0,0,1Z"
              />
              <path
                d="M9.71,17.18a2.587,2.587,0,0,0,1.12-.65l9.54-9.54a1.75,1.75,0,0,0,0-2.47l-.94-.93a1.788,1.788,0,0,0-2.47,0L7.42,13.12a2.473,2.473,0,0,0-.64,1.12L6.04,17a.737.737,0,0,0,.19.72.767.767,0,0,0,.53.22Zm.41-1.36a1.468,1.468,0,0,1-.67.39l-.97.26-1-1,.26-.97a1.521,1.521,0,0,1,.39-.67l.38-.37,1.99,1.99Zm1.09-1.08L9.22,12.75l6.73-6.73,1.99,1.99Zm8.45-8.45L18.65,7.3,16.66,5.31l1.01-1.02a.748.748,0,0,1,1.06,0l.93.94A.754.754,0,0,1,19.66,6.29Z"
              />
            </g>
          </g>
        </svg>
      </button>
    </article>
    <aside
      className="w-full md:w-[40%] h-fit mt-20 md:mt-0 "
    >
      <h2
        className="text-2xl font-medium text-center mb-4"
      >
        Read Also
      </h2>
      <div
        className="flex blog-aside flex-wrap justify-center gap-10 w-full blog-wrapper"
      >
        <div
          aria-hidden="true"
          className="rounded blogItem
    overflow-hidden w-[60%]
    flex
 cursor-pointer
    flex-col
   justify-between
   brightness-100
   transition-all
   ease-in
   duration-500
   hover:brightness-75
     shadow-lg"
          onClick={[Function]}
        >
          <div
            className="grow-0"
          >
            <img
              alt=""
              className="w-full aspect-[4/1]  object-cover"
            />
          </div>
          <div
            className="grow flex flex-col justify-between"
          >
            <div
              className="px-6 py-4"
            >
              <div
                className="font-bold text-xl mb-2"
              >
                 
              </div>
              <p
                className="text-gray-700 text-base blog-text"
              />
            </div>
            <div
              className="px-6 pt-4 pb-2 flex items-center justify-between"
            >
              <div
                className="flex items-center"
              >
                <img
                  alt=""
                  className="rounded-full !w-[40px] aspect-square md:w-[70px] h-13"
                  src="man.png"
                />
                <p
                  className="text-xl inline-block"
                >
                   
                </p>
              </div>
              <span
                className="flex items-center justify-center z-30"
              >
                <svg
                  className="text-rose text-2xl mr-1 "
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
                  />
                </svg>
                <span
                  className="text-lg font-medium"
                />
              </span>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          className="rounded blogItem
    overflow-hidden w-[60%]
    flex
 cursor-pointer
    flex-col
   justify-between
   brightness-100
   transition-all
   ease-in
   duration-500
   hover:brightness-75
     shadow-lg"
          onClick={[Function]}
        >
          <div
            className="grow-0"
          >
            <img
              alt=""
              className="w-full aspect-[4/1]  object-cover"
            />
          </div>
          <div
            className="grow flex flex-col justify-between"
          >
            <div
              className="px-6 py-4"
            >
              <div
                className="font-bold text-xl mb-2"
              >
                 
              </div>
              <p
                className="text-gray-700 text-base blog-text"
              />
            </div>
            <div
              className="px-6 pt-4 pb-2 flex items-center justify-between"
            >
              <div
                className="flex items-center"
              >
                <img
                  alt=""
                  className="rounded-full !w-[40px] aspect-square md:w-[70px] h-13"
                  src="man.png"
                />
                <p
                  className="text-xl inline-block"
                >
                   
                </p>
              </div>
              <span
                className="flex items-center justify-center z-30"
              >
                <svg
                  className="text-rose text-2xl mr-1 "
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
                  />
                </svg>
                <span
                  className="text-lg font-medium"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </aside>
  </div>
  <div
    className="Toastify"
  />
</div>
`;
